src/messaging/ReliableMessageManager.cpp

```diff
@@ -413,18 +459,31 @@ CHIP_ERROR ReliableMessageManager::SendFromRetransTable(RetransTableEntry * entr
 
     if (rc)
     {
-        // Locally store the start and length;
-        uint8_t * p  = entry->msgBuf->Start();
-        uint16_t len = entry->msgBuf->DataLength();
+        System::PacketBufferHandle msgBuf;
+
+        msgBuf.Adopt(entry->msgBuf);
```

Comment:
Why not just store a `PacketBufferHandle` in the table instead of a raw pointer?  Then here you would do:
```
  System::PacketBufferHandle msgBuf = entry->msgBuf.Retain();
```

Reply:
Same as above, PacketBufferHandle can only be moved, the PacketBufferHandle.mBuffer will become nullptr after each send,  what we use for the following re-transmission?

Reply:
> Same as above, PacketBufferHandle can only be moved,

Again.  If `entry->msgBuf` is a PacketBufferHandle, and you do:
```
  System::PacketBufferHandle msgBuf = entry->msgBuf.Retain();
```
You can now move out of `msgBuf` without affecting the fact that `entry` still owns a reference to the packet buffer.

Reply:
Acked